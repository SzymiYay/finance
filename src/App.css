#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

body {
  font-family: Arial, Helvetica, sans-serif;
  margin: 40px;
  background: #f9f9f9;
}

h1 {
  text-align: center;
}

table {
  border-collapse: collapse;
  margin: 20px auto;
  width: 60%;
  background: white;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
}

th,
td {
  border: 1px solid #ddd;
  padding: 10px;
  text-align: center;
}

th {
  background: #0077cc;
  color: white;
}

tr:nth-child(even) {
  background: #f2f2f2;
}

#chart-container {
  width: 70%;
  margin: 40px auto;
}
