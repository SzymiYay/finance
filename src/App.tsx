import './App.css'
import MomentumChart from './components/MomentumChart'
import ResultsTable from './components/ResultsTable'

function App() {
  return (
    <>
      <>
        <h1>ETF Momentum (%)</h1>
        <ResultsTable />
        <MomentumChart />
      </>
    </>
  )
}

export default App
