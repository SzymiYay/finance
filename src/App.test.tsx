import App from './App'
import { render, screen } from '@testing-library/react'

describe('App component', () => {
  test('renders heading', () => {
    render(<App />)
    expect(
      screen.getByRole('heading', { name: /ETF Momentum/i })
    ).toBeInTheDocument()
  })

  test('App matches snapshot', () => {
    const { asFragment } = render(<App />)
    expect(asFragment()).toMatchSnapshot()
  })
})
