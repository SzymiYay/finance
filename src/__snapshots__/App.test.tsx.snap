// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > App matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    ETF Momentum (%)
  </h1>
  <table
    id="results-table"
  >
    <thead>
      <tr>
        <th>
          ETF
        </th>
        <th>
          12M Momentum (%)
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <p>
    Loading chart...
  </p>
</DocumentFragment>
`;
